### ISSUE TITLE
Search in file browser

### ISSUE THREAD

**Issue by AbdealiJK** [jupyterlab/jupyterlab#818]
It would be really nice to have a search feature for the file browser, where I can type the partial name of the file I want to open and it filters the list based on that.

Currently I can use the browser's search to highlight the partial filename, but it interferes with the rest of the content also

It can probably fit right into the space near the home button in the file browser

**Comment by AbdealiJK** [comment]
Just found that a similar feature has been suggested in https://github.com/jupyter/jupyterlab/issues/213 (Point 4) but without the search box (You simply type the text you want to find in that with a timeout)
A search box may be a better interface though - although that's debatable.

**Comment by blink1073** [comment]
Hi @AbdealiJK, this is certainly something worth exploring, but difficult to get right.

**Comment by jasongrout** [comment]
The command palette filtering provides a nice guide for how it might be done, though. By the way, SageMathCloud has this filter mechanism.

**Comment by ellisonbg** [comment]
Yeah, I think this would be nice to have eventually.

**Comment by jasongrout** [comment]
@parente recently experimented with a JupyterLab plugin for doing a 'quick open' feature: https://github.com/parente/jupyterlab-quickopen

**Comment by dclong** [comment]
I think I also I opened similar issue before. I tried jupyterlab-quickopen but I found it to have some performance issues when the work directory containing a large number of files. Unfortunately, the situation when there are lots of file under the directory is when jupyterlab-quickopen can help the users the most. 

As an alternative, I fall back to the plain old command `find` which serves the search purpose very well. The only limitation is that JupyterLab does not support opening files from terminal at this time. 
https://github.com/jupyterlab/jupyterlab/issues/5056
